From 4237fe857348c69f3419eabe5813f809d678e1fa Mon Sep 17 00:00:00 2001
From: Vasily Khoruzhick <anarsoul@gmail.com>
Date: Thu, 23 Nov 2017 20:19:05 -0800
Subject: [PATCH 116/134] lima: use rb_root_cached instead of rb_root

All users of inteval tree are required to use rb_root_cached
since f808c13f. Switch to rb_root_cached to fix lima compilation
with 4.15

Signed-off-by: Vasily Khoruzhick <anarsoul@gmail.com>
---
 drivers/gpu/drm/lima/lima_vm.c | 5 +++--
 drivers/gpu/drm/lima/lima_vm.h | 2 +-
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/lima/lima_vm.c b/drivers/gpu/drm/lima/lima_vm.c
index c65acbf..7f36260 100644
--- a/drivers/gpu/drm/lima/lima_vm.c
+++ b/drivers/gpu/drm/lima/lima_vm.c
@@ -148,7 +148,7 @@ struct lima_vm *lima_vm_create(struct lima_device *dev)
 		return NULL;
 
 	vm->dev = dev;
-	vm->va = RB_ROOT;
+	vm->va = RB_ROOT_CACHED;
 	mutex_init(&vm->lock);
 	kref_init(&vm->refcount);
 
@@ -178,8 +178,9 @@ void lima_vm_release(struct kref *kref)
 		}
 	}
 
-	if (!RB_EMPTY_ROOT(&vm->va))
+	if (!RB_EMPTY_ROOT(&vm->va.rb_root)) {
 		dev_err(vm->dev->dev, "still active bo inside vm\n");
+	}
 
 	for (i = 0; (vm->pd.dma & LIMA_PAGE_MASK) && i < LIMA_PAGE_ENT_NUM; i++) {
 		if (vm->pts[i].cpu) {
diff --git a/drivers/gpu/drm/lima/lima_vm.h b/drivers/gpu/drm/lima/lima_vm.h
index fb7846b..7d07085 100644
--- a/drivers/gpu/drm/lima/lima_vm.h
+++ b/drivers/gpu/drm/lima/lima_vm.h
@@ -44,7 +44,7 @@ struct lima_vm {
 	struct lima_device *dev;
 
 	/* tree of virtual addresses mapped */
-	struct rb_root va;
+	struct rb_root_cached va;
 
         struct lima_vm_page pd;
 	struct lima_vm_page pts[LIMA_PAGE_ENT_NUM];
-- 
2.0.1

